src/playground/sample_data_ingestion/jobs/startup_job.py

```diff
@@ -0,0 +1,145 @@
+# opensearch-py==2.0.0
+from opensearchpy import OpenSearch
+
+# Standard libraries
+import sys
+from os import environ, remove, listdir, path
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+from sample_data_indices.sample_data_indices import SampleDataIndex
+from sample_data_plugins.sample_data_plugin import AnomalyDetection, Alerting
+from datetime import date, timedelta, datetime
+from shutil import copyfileobj
+from time import sleep
+from math import ceil
+from json import load
+import gzip
+
+# Array in which unzipped files will be removed
+file_removal_array = []
+
+
+# Utility function for unzipping files
+def unzip(filename):
+  if type(filename) is not str or (type(filename) is str and ".json" not in filename):
+      print("Payload is not a filename or is not a .json file")
+      return None
+  if type(filename) is str and ".gz" not in filename:
+      print("Payload is already unzipped")
+      return filename
+  with gzip.open(filename, 'rt') as fin:
+      with open(filename.split(".gz")[0], 'wb') as fout:
+          copyfileobj(fin, fout)
+  file_removal_array.append(filename.split(".gz")[0])
+  return filename.split(".gz")[0]
```

Comment:
same as refresh_job.py, if user add *.tar.gz, what's happening here? thanks

Reply:
Done, I added logic that extracts files from *tar.gz files before the tooling scans for files. The actual logic is located in sample_data_commons/utils.py.